lib/baby-braspag/credit_card.rb

```diff
@@ -186,25 +185,53 @@ def self.just_click_shop(params = {})
       ProtectedCreditCard.just_click_shop(params)
     end
 
-    def self.info_url
-      connection = Braspag::Connection.instance
-      connection.braspag_url + (connection.production? ? PRODUCTION_INFO_URI : HOMOLOGATION_INFO_URI)
-    end
+    class << self
+      private
 
-    def self.authorize_url
-      Braspag::Connection.instance.braspag_url + AUTHORIZE_URI
-    end
+      def check_params(params)
+        super
 
-    def self.capture_url
-      Braspag::Connection.instance.braspag_url + CAPTURE_URI
-    end
+        [:customer_name, :holder, :card_number, :expiration, :security_code, :number_payments, :type].each do |param|
+          raise IncompleteParams if params[param].nil?
+        end
 
-    def self.partial_capture_url
-      Braspag::Connection.instance.braspag_url + PARTIAL_CAPTURE_URI
-    end
+        raise InvalidHolder if params[:holder].to_s.size < 1 || params[:holder].to_s.size > 100
+
+        matches = params[:expiration].to_s.match /^(\d{2})\/(\d{2,4})$/
+        raise InvalidExpirationDate unless matches
 
-    def self.cancellation_url
-      Braspag::Connection.instance.braspag_url + CANCELLATION_URI
+        year = matches[2].size == 4 ? matches[2] : "20#{matches[2]}"
+        raise if !Date.valid_date?(year.to_i, matches[1].to_i, 1)
```

Comment:
Shouldn't we raise `InvalidExpirationDate` here as well? Even though this validation here is more semantic, whereas the former is more syntactic.